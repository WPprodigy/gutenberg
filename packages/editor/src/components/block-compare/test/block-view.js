/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import BlockView from '../block-view';

describe( 'BlockView', () => {
	const onAction = jest.fn();
	const wrapper = shallow( <BlockView title="title" rawContent="raw" renderedContent="render" action={ onAction } actionText="action" className="class" /> );

	afterEach( () => {
		onAction.mockClear();
	} );

	test( 'should match snapshot', () => {
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should call onAction when button is pressed', () => {
		wrapper.find( 'Button' ).simulate( 'click' );

		expect( onAction ).toHaveBeenCalledTimes( 1 );
	} );
} );
