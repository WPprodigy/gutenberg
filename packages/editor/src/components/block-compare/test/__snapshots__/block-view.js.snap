// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockView should match snapshot 1`] = `
<div
  className="class"
>
  <h1
    className="components-modal__header-heading"
  >
    title
  </h1>
  <div
    className="editor-block-compare__html"
  >
    raw
  </div>
  <div
    className="editor-block-compare__preview edit-post-visual-editor"
  >
    render
  </div>
  <div
    className="editor-block-compare__action"
  >
    <Button
      isLarge={true}
      onClick={[MockFunction]}
    >
      action
    </Button>
  </div>
</div>
`;
